src/components/calcite-tree-item/calcite-tree-item.tsx

```diff
@@ -50,16 +50,22 @@ export class CalciteTreeItem {
 
   @Watch("expanded")
   expandedHandler(newValue: boolean): void {
-    const items = getSlotted<HTMLCalciteTreeItemElement>(this.el, "children", {
-      all: true,
-      selector: "calcite-tree-item"
-    });
+    // handle newly appended items on parents
+    if (!newValue && this.parentTreeItem) {
+      const { expanded } = this.parentTreeItem;
```

Comment:
Does this logic apply outside of newly appended tree items? If not, maybe this could be moved into `connectedCallback` to clearly separate this.

Reply:
Good call Franco, I'll take this approach.